### ISSUE TITLE
regression in v3 – dates outside ranges in `selectedDates` get the attribute `data-vc-date-selected="middle"`

### ISSUE THREAD

**Issue by johanrd** [uvarov-frontend/vanilla-calendar-pro#319]
"vanilla-calendar-pro": "^3.0.0-beta.12",

Steps to rerpoduce:
1) Init a calendar with the following settings:

```
selectedDates: ['2024-10-01:2024-10-04', '2024-10-15:2024-10-18']
selectedTheme: 'system',
selectionDatesMode: 'multiple-ranged',
```
2. See that the dates between the two ranges are selected with the attribute: `data-vc-date-selected="middle"`:

| Actual  |      Expected      |
|----------|:-------------:|
| <img width="265" alt="Screenshot 2024-11-02 at 14 35 49" src="https://github.com/user-attachments/assets/d3c35f1f-0243-45b0-a84b-b2769b6aa25c"> | <img width="271" alt="Screenshot 2024-11-02 at 14 47 30" src="https://github.com/user-attachments/assets/7a5ceb4a-67da-4ee9-9f35-e2ae7e4901e9"> |

<img width="1259" alt="Screenshot 2024-11-02 at 14 39 43" src="https://github.com/user-attachments/assets/9ad28e81-8f86-466d-a08a-6911d9e79632">

## Expected:
Dates 2024-10-05 – 2024-10-14 to not have the attribute `data-vc-date-selected="middle"`:

Would also expect something like 
`data-vc-date-selected'last-in-range'` on `2024-10-04`
`data-vc-date-selected'first-in-range'` on `2024-10-15`,
 according to https://github.com/uvarov-frontend/vanilla-calendar-pro/issues/313

**Comment by uvarov-frontend** [comment]
Try turning on false for this parameter: `enableEdgeDatesOnly: false`. Let me know.

**Comment by uvarov-frontend** [comment]
And by the way, you cannot specify two ranges in `selectionDatesMode: 'multiple-ranged'`. In theory, you can assign this during initialization, but not using manual selection.

**Comment by johanrd** [comment]
> Try turning on false for this parameter: `enableEdgeDatesOnly: false`. Let me know.

thanks, yes! setting `enableEdgeDatesOnly: false` works!  

> And by the way, you cannot specify two ranges in `selectionDatesMode: 'multiple-ranged'`. In theory, you can assign this during initialization, but not using manual selection.

Ah, ok, I understand that my use case is an edge case. For info: I use it as a read-only viewer of 'visited dates', think of it like a historical view of check-ins- and outs – like gantt-chart view in a calendar.

**Comment by uvarov-frontend** [comment]
> > Try turning on false for this parameter: `enableEdgeDatesOnly: false`. Let me know.
> 
> thanks, yes! setting `enableEdgeDatesOnly: false` works!
> 
> > And by the way, you cannot specify two ranges in `selectionDatesMode: 'multiple-ranged'`. In theory, you can assign this during initialization, but not using manual selection.
> 
> Ah, ok, I understand that my use case is an edge case. For info: I use it as a read-only viewer of 'visited dates', think of it like a historical view of check-ins- and outs – like gantt-chart view in a calendar.

I need to set `enableEdgeDatesOnly: false` by default then. Thanks for the report.

For reading, your option is ok.

**Comment by afonic** [comment]
Hey! Congrats on v3!

Sorry to comment on a closed issue but I think it's relevant.

I'm using the Calendar in `multiple-ranged` and and I'm trying to display the date range on hover and on select even though some days are disabled. I want the user to select ranges only from Saturday to Saturday so I set `disableWeekdays: [0,1,2,3,4,5]`. This way the user can correctly only select weeks starting and ending on Saturdays but I also need to display the range in the calendar somehow as if the days between were not disabled. See this example using another calendar: https://archonyachting.com/

I played with `enableEdgeDatesOnly` but it doesn't seem to add `date-selected` or any other property. 

Is this possible via config at the moment?
